### PR TITLE
feat: Add tinymist

### PR DESCRIPTION
**Pull request by Parsifa1**
Add a typst lsp.

### CODE REVIEW

File: packages/tinymist/package.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+homepage: https://github.com/Myriad-Dreamin/tinymist
+licenses:
+  - Apache-2.0
+languages:
+  - Typst
+categories:
+  - LSP
+
+source:
+  id: pkg:github/Myriad-Dreamin/tinymist@v0.10.2
```
Comment (Conarius) [inline packages/tinymist/package.yaml:14]: There already is a newer version available, would advise to update it

Comment (Andrew15-5) [reply packages/tinymist/package.yaml:14]: Wouldn't the bot automatically update it anyway?

Comment (Andrew15-5) [reply packages/tinymist/package.yaml:14]: Wouldn't the bot automatically update it anyway?

File: packages/tinymist/package.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    - target: win_x64
+      file: tinymist-win32-x64
+      bin: tinymist-win32-x64.exe
+    - target: win_arm64
+      file: tinymist-win32-arm64
+      bin: tinymist-win32-arm64.exe
+
+bin:
+  tinymist: "{{source.asset.bin}}"
+# test workflow
```
[the remark this answers is not in the document]
Comment (Parsifa1) [reply packages/tinymist/package.yaml:37]: Oh, here's the comment from the file I referenced earlier, I'll delete it

**Review by williamboman — APPROVED** [review]
Thanks!

**Pull request merged**